src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
```

Comment:
Use a destruction for `action`
```suggestion
  for (const action of actions) {
    const { type, ........ } = action;
  
    if (type === 'addProperties') {
```